app/api/src/docker/common/utils.js

```diff
@@ -1,6 +1,19 @@
 const { exec } = require('child_process')
 const { promises } = require('fs')
 const { writeFile } = promises
+const { createHash } = require('crypto')
+
+const CONSOLE_MESSAGE_KEY = 'console-message-b64'
+function putConsoleMessageInMetadata(consoleMessage) {
```

Comment:
I'm storing whatever the cad package printed to the terminal when it made the part in metadata, so that I can still display this when returning the cached assets. s3 seems to be pretty sensitive about special characters that can be put in s3 so base64 to the rescue.

Reply:
Ended up being a bad idea as only 2kb can be store as metadata, and it's definitely possible for logs to exceed this.